RestaurantReservation.API/Controllers/EmployeesController.cs

```diff
@@ -0,0 +1,226 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantReservation.API.Models.Employees;
+using RestaurantReservation.Db.Interfaces;
+using RestaurantReservation.Db.Models.Entities;
+using System.Text.Json;
+
+namespace RestaurantReservation.API.Controllers
+{
+    [Route("api/employees")]
+    [ApiController]
+    [Authorize]
+    public class EmployeesController : ControllerBase
+    {
+        private readonly IEmployeeRepository _employeeRepository;
+        private readonly IRestaurantRepository _restaurantRepository;
+        private readonly IOrderRepository _orderRepository;
+        private readonly IMapper _mapper;
+        private const int MaxPageSize = 5;
+
+        public EmployeesController(IEmployeeRepository employeeRepository, IRestaurantRepository restaurantRepository, IOrderRepository orderRepository, IMapper mapper)
+        {
+            _employeeRepository = employeeRepository;
+            _restaurantRepository = restaurantRepository;
+            _orderRepository = orderRepository;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Retrieves all employees with pagination.
+        /// </summary>
+        /// <param name="pageNumber">The number of the page to retrieve.</param>
+        /// <param name="pageSize">The number of items per page.</param>
+        /// <returns>A list of employee DTOs.</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult<IEnumerable<EmployeeDto>>> GetEmployees(int pageNumber = 1, int pageSize = 10)
+        {
+            if (pageNumber < 1 || pageSize < 1)
```

Comment:
this logic so far was repeated twice in controllers, it can be moved to a common class